iot-hub/Samples/device/DeviceReconnectionSample/readme.md

```diff
@@ -2,6 +2,113 @@
+};
+
+await deviceClient.SendEventAsync(message);
+```
+
+### Receive cloud to device telemetry (using the polling API) and complete the message:
+
+```csharp
+// This snippet shows you how to call the API for receiving telemetry sent to your device client.
+// In order to ensure that your client is resilient to disconnection events and exceptions, refer to https://github.com/Azure-Samples/azure-iot-samples-csharp/blob/master/iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs.
+using Message receivedMessage = await deviceClient.ReceiveAsync();
+if (receivedMessage == null)
+{
+    Console.WriteLine("No message received; timed out.");
+    return;
+}
+
+string messageData = Encoding.ASCII.GetString(receivedMessage.GetBytes());
+var formattedMessage = new StringBuilder($"Received message: [{messageData}]\n");
+
+// User set application properties can be retrieved from the Message.Properties dictionary.
+foreach (KeyValuePair<string, string> prop in receivedMessage.Properties)
+{
+    formattedMessage.AppendLine($"\tProperty: key={prop.Key}, value={prop.Value}");
+}
+
+// System properties can be accessed using their respective accessors.
+formattedMessage.AppendLine($"\tMessageId: {receivedMessage.MessageId}");
+
+Console.WriteLine(formattedMessage.ToString());
+await deviceClient.CompleteAsync(receivedMessage);
+```
+
+### Receive cloud to device telemetry (using the callback) and complete the message:
+
+```csharp
+// This snippet shows you how to call the API for receiving telemetry sent to your device client.
+// In order to ensure that your client is resilient to disconnection events and exceptions,
+// refer to https://github.com/Azure-Samples/azure-iot-samples-csharp/blob/master/iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs.
+private async Task OnC2dMessageReceived(Message receivedMessage, object userContext)
+{
+    string messageData = Encoding.ASCII.GetString(receivedMessage.GetBytes());
+    var formattedMessage = new StringBuilder($"Received message: [{messageData}]\n");
+
+    // User set application properties can be retrieved from the Message.Properties dictionary.
+    foreach (KeyValuePair<string, string> prop in receivedMessage.Properties)
+    {
+        formattedMessage.AppendLine($"\tProperty: key={prop.Key}, value={prop.Value}");
+    }
+
+    // System properties can be accessed using their respective accessors.
+    formattedMessage.AppendLine($"\tMessageId: {receivedMessage.MessageId}");
+
+    Console.WriteLine(formattedMessage.ToString());
+    await deviceClient.CompleteAsync(receivedMessage);
+}
+
+// Subscribe to the receive message API.
+await deviceClient.SetReceiveMessageHandlerAsync(OnC2dMessageReceived, deviceClient);
+
+// Once you are done receiving telemetry messages sent to your device client,
+// you can unsubscribe from the receive callback by setting a null handler.
+await deviceClient.SetReceiveMessageHandlerAsync(null, deviceClient);
```

Comment:
will be good to add this in sample code as well.

Reply:
sure, i'll add that